dx-cli 0.7.0 Fails to build new project on Windows

**Problem**

Running `dx serve` fails when performing the link step when running on Windows 11.

**Steps To Reproduce**

Steps to reproduce the behavior:

- create a new barebones desktop project using `dx cli`
- `cd` to new project directory
- run `dx serve`

**`dx serve` output**
```
21:30:49 [cargo] error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.44.35207\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\kgret\\AppData\\Local\\Temp\\rustcrtaRoj\\symbols.o" "<94 object files omitted>" "D:\\dev\\projects\\dx7-fresh\\target\\x86_64-pc-windows-msvc\\desktop-dev\\deps/{libdioxus-6bdf0e06f4a1792b,libmanganis-8023a63a843c5eaf,libmanganis_core-793d24f796165391,libconst_serialize-7408d10d26b20274,libdioxus_config_macros-99389103d46c99c6,libdioxus_desktop-0e2baa0eeb603edb,libdioxus_asset_resolver-8d924e10fda0ba5f,libinfer-a1668b4a3bb82c92,libcfb-f511abb242d485cc,libuuid-1c44aede8fad6ec7,libbyteorder-24185b184e579792,librfd-19aa345a093875c7,libsubtle-7e1f383a13bb79a8,libwebbrowser-b5aba873d015e962,libtray_icon-cd0be97a4f9d9ca8,libglobal_hotkey-0180f0e1cdafd0ad,libwindows_sys-08ed028309b9ba2a,libwindows_targets-a5809ccc6ad20fba,libbase64-620581ffff1c9099,libmuda-86974b9a5653103b,libwindows_sys-f7b406e4407c7dca,libwindows_targets-68e8e92e4ffd9f6d,libtokio-f5cc1870c20fd913,libdioxus_interpreter_js-508b432831310068,libsledgehammer_utils-25cb28944cae7e77,librustc_hash-baa7e05cb589c322,libsledgehammer_bindgen-7ceed96dfdffc4cf,libwry-fbafc2c68a01fdf2,libdunce-1c5ba781c61c710c,libcookie-ab7f30b20906eef5,libtime-6f83eb6139c8416a,libtime_core-2c430ba2474f1ef5,libnum_conv-dba7b3532eab1ed2,libderanged-b33fa5b5db28b7ba,libpowerfmt-0abd3f0f25b93118,libwebview2_com-0f0f7cbd7593fadf,libwebview2_com_sys-e7cbb64bbfe963ba,libtao-d31355708c6e9ccf,libwindows_version-b5908b0e87fd1ed1,libcrossbeam_channel-325b69a38ab4fcad,libcrossbeam_utils-f5a6044b592f202f,libwindows-146662e2d906c15a,libwindows_core-38246eb8dec585eb,libwindows_strings-88f35d55d41e8d59,libwindows_result-c6aac9e443413fa8,libwindows_link-dba2e5a4c1995cf4,libdpi-5e51f4d09632f1f3,libraw_window_handle-4b43698ea62e3012,libraw_window_handle-9b37df4c7b3347f1,libdioxus_router-cb4e6c121bf67ed6,liburl-12445b20e3bc90bd,libidna-8da384fba9988caa,libutf8_iter-071eba5ccb46a690,libidna_adapter-a86a682339acee16,libicu_properties-23f2f91c40b7ccfc,libicu_properties_data-5889f13650533ff1,libicu_normalizer-4e14556b40df811e,libicu_normalizer_data-bdc57fd686378f01,libicu_collections-7cddb9e1739c5749,libpotential_utf-19ceee790132816b,libicu_provider-4863e0ed7ff68a5e,libicu_locale_core-2a399f841b1d28c0,libtinystr-190262fcb86f6d69,liblitemap-cbbcc5f78acffa35,libwriteable-0d23a48cb3c7c021,libzerovec-2de6a2e6e38dd941,libzerotrie-167c385ce5e8d29a,libyoke-b1efa767cdc5fb91,libstable_deref_trait-d4eaca71097d4569,libzerofrom-01ed6a19f20c8431,libform_urlencoded-4071efd1177c355a,libpercent_encoding-d3746fd9fc821547,libdioxus_devtools-5b4db8375730dd4a,libtungstenite-07ae3e81997155e4,libdata_encoding-67d4d43863e71285,librand-a4d4ede3fa55071a,librand_chacha-de4899c459a37636,libppv_lite86-8c30e0da8d8b3369,libzerocopy-8d604cb5e0d32915,librand_core-d23563e312e6cab6,libgetrandom-9e284f63ca3032c8,libnative_tls-0144323f671c0850,libschannel-bbcd7b25d51c20e8,libwindows_sys-79c660b47289c56b,libutf8-14576e5e99a19b53,libsha1-51be2701a0548c07,libcpufeatures-3fee921f4d1a1e1f,libdigest-508ddb7c7f4b5ddd,libblock_buffer-628ef60ebc862b0e,libcrypto_common-ce0f3d086f9e0c7c,libgeneric_array-3d1e5a1536cf6e35,libtypenum-99571f7c5594e0e0,libhttparse-0ad4b47a170486db,liblog-0910c1ab6f9fe111,libhttp-bb9481f24b615e02,libfnv-b7ceaa700d7eba91,libdioxus_devtools_types-1e67c1acd03673f4,libdioxus_logger-7fe0796c8bbb45f7,libdioxus_cli_config-fb8a4d7c17a310b6,libtracing_subscriber-fd59373b18c4ddb6,libsharded_slab-0d7fc82d89e410fa,liblazy_static-747bb9ad9b2517a7,libmatchers-3c5d467db78121fd,libregex_automata-40d6a8b07540dc55,libregex_syntax-64684a33f30993ae,libthread_local-8599014f35c65db8,libdioxus_history-011b4f18475dd5ca,libdioxus_document-5aadf636383dd269,libdioxus_html-e70e5eca41e7d83b,libserde_json-fe7ce667e5dae189,libmemchr-5657c6861b8935aa,libitoa-7db479836050a3ed,libryu-61682fa67e8eea7d,libenumset-4913901e0ff16b79,libeuclid-0350e8c436474b43,libnum_traits-19fd9b74c2630b77,libbytes-917b73243ac6492f,libkeyboard_types-91ffc66ee9f3e0c8,libbitflags-ce3f8e72974d4b2e,libunicode_segmentation-3666fd5138c43ec2,libdioxus_stores-8e14cb36f75ff84c,libdioxus_hooks-b8c37a778d675c63,libdioxus_signals-1351df092115e6a8,libwarnings-438f985d6cbeec25,libpin_project-2874e9fc00ce8762,libdioxus_core-2cb741bbfa740b83,libsubsecond-df5270e0a7c0d66b,liblibloading-1e7d184ee3d6da18,libthiserror-4aea380554158ff2,libsubsecond_types-adfd8c7ad2f7dc2e,liblongest_increasing_subsequence-c3e8772e6f241d08,libconst_format-46b24a46b8cf731d,libanyhow-90e121162f241198,libslotmap-3fc8f010cb33a51e,libfutures_channel-526e43d8adc8481d,libserde-e450456189c00d2b,libserde_core-2b91408a3c1f39ae,libdioxus_core_types-efdf3183c059eaaa,libgenerational_box-a46452127dc06462,libtracing-742d606dc3118a69,libtracing_core-31861b80de73a439,libonce_cell-924209fb4e1b9858,libparking_lot-7804a68e9bb01ac9,libparking_lot_core-2a2e1df3abe5c152,libwindows_link-d08cb4c3645f2300,libcfg_if-3f883b937b80765f,libsmallvec-6f0e6abccf2caeeb,liblock_api-e34a4365d3a8dc52,libscopeguard-bb339d256641f19d,libfutures_util-7d1c0be30a233247,libslab-9d5ec9f3780c9faf,libpin_project_lite-65600a047c864d39,libfutures_task-87bcf67fcdc3540  = note: some arguments are omitted. use `--verbose` to show all linker arguments5285004b51e2b}.rlib" "<sysroot>\\lib\\
  = note: msvcrt.lib(exe_winmain.obj) : error LNK2019: unresolved external symbol WinMain referenced in function "int __          d:\dev\projects\dx7-fresh\target\x86_64-pc-windows-msvc\desktop-dev\deps\dx7_fresh.exe : fatal error LNK1120:           ed externals␍-*}.rlib" "shlwapi.lib" "C:\\Users\\kgret\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib\\windows.0.52.0.lib" "C:\\Users\\kgret\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.53.1\\lib\\windows.0.53.0.lib" "advapi32.lib" "kernel32.lib" "kernel32.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "dbghelp.lib" "/defaultlib:msvcrt" "/NXCOMPAT" "/LIBPATH:d:\\dev\\projects\\dx7-fresh\\target\\x86_64-pc-windows-msvc\\desktop-dev\\build\\webview2-com-sys-cdf25205542486ab\\out\\x64" "/LIBPATH:C:\\Users\\kgret\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.52.6\\lib" "/LIBPATH:C:\\Users\\kgret\\.cargo\\registry\\src\\index.crates.io-1949cf8c6b5b557f\\windows_x86_64_msvc-0.53.1\\lib" "/OUT:d:\\dev\\projects\\dx7-fresh\\target\\x86_64-pc-windows-msvc\\desktop-dev\\deps\\dx7_fresh.exe" "/OPT:REF,NOICF" "/DEBUG" "/PDBALTPATH:%_PDB%" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:<sysroot>\\lib\\rustlib\\etc\\libstd.natvis" "/NATVIS:C:\\Users\\kgret\\AppData\\Local\\Temp\\rustcrtaRoj\\dx7_fresh-0.natvis" "/NATVIS:C:\\Users\\kgret\\AppData\\Local\\Temp\\rustcrtaRoj\\dx7_fresh-1.natvis" "/NATVIS:C:\\Users\\kgret\\AppData\\Local\\Temp\\rustcrtaRoj\\dx7_fresh-2.natvis" "/SUBSYSTEM:WINDOWS"
21:30:49 [dev] Build failed: cargo build finished with errors for target: dx7-fresh [x86_64-pc-windows-msvc]
```

**Environment:**

- Dioxus version:  0.7.0
- Rust version:   rustc 1.91.0 (f8297e351 2025-10-28) 
- OS info:         Windows 11
- App platform:    desktop

--- a/packages/cli/src/build/request.rs
+++ b/packages/cli/src/build/request.rs
@@ -908,7 +908,7 @@ impl BuildRequest {
             ]);
         }
 
-        // On windows, we pass /SUBSYSTbEM:WINDOWS to prevent a console from appearing
+        // On windows, we pass /SUBSYSTEM:WINDOWS to prevent a console from appearing
         if matches!(bundle, BundleFormat::Windows)
             && !rustflags
                 .flags
@@ -922,6 +922,11 @@ impl BuildRequest {
             rustflags
                 .flags
                 .push(format!("-Clink-arg=/SUBSYSTEM:{}", subsystem));
+            // We also need to set the entry point to mainCRTStartup to avoid windows looking
+            // for a WinMain function
+            rustflags
+                .flags
+                .push("-Clink-arg=/ENTRY:mainCRTStartup".to_string());
         }
 
         // Make sure we set the sysroot for ios builds in the event the user doesn't have it set

